src/NuGet.Core/NuGet.Frameworks/FrameworkRuntimePair.cs

```diff
@@ -7,7 +7,10 @@
 
 namespace NuGet.Frameworks
 {
-    public class FrameworkRuntimePair : IEquatable<FrameworkRuntimePair>, IComparable<FrameworkRuntimePair>
+    /// <remarks>
+    /// Immutable.
+    /// </remarks>
+    public sealed class FrameworkRuntimePair : IEquatable<FrameworkRuntimePair>, IComparable<FrameworkRuntimePair>
```

Comment:
Why is this not being detected as a breaking chnage by the analyzer? I thought https://github.com/NuGet/NuGet.Client/pull/5234 might fix that. 

@zivkan any ideas?

Reply:
There are a lot of things the current public API analyzers don't catch. See https://github.com/dotnet/roslyn-analyzers/issues/3047 for more info. I think our best way of addressing that would be https://github.com/dotnet/roslyn-analyzers/issues/3901.